### PR TITLE
raidboss: fix ASS Silkie triggers

### PR DESCRIPTION
**Pull request by xiashtra**
Attempt to fix ASS Silkie triggers for first and third Soaping Spree and third Slippery Soap.

Addresses #4992
cc: @Legends0

### CODE REVIEW

File: ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts [hunk 1]
```diff
@@ -126,13 +126,22 @@ const triggerSet: TriggerSet<Data> = {
...
+      run: (data, matches) => data.suds = matches.effectId,
+    },
+    {
+      id: 'ASS Suds Lose',
+      // CE1 Bracing Suds (Wind / Donut)
+      // CE2 Chilling Suds (Ice / Cardinal)
+      // CE3 Fizzling Suds (Lightning / Intercardinal)
+      type: 'LosesEffect',
+      netRegex: { effectId: 'CE[1-3]', target: 'Silkie', capture: false },
+      run: (data, _matches) => data.suds = '',
```
Comment (quisquous) [inline ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts:144]:
```suggestion
      run: (data, _matches) => delete data.suds,
```

Can you use `undefined` as unset rather than empty string?

**Comment by Legends0** [comment]
Hmm, I see in the timeline, it's Eastern Ewer casting the Slippery Soap:
204.4 "--sync--" sync / 1[56]:[^:]*:Eastern Ewer:79FB:/
209.4 "Slippery Soap" sync / 1[56]:[^:]*:Silkie:775E:/

So alternatively the previous trigger just needed ['Silkie, 'Eastern Ewer'].

**Comment by Legends0** [comment]
Looks like I also missed that there aren't suds casts for Soaping Spree, so a finer check on the debuff is necessary here.

**Comment by xiashtra** [comment]
> Hmm, I see in the timeline, it's Eastern Ewer casting the Slippery Soap: 204.4 "--sync--" sync / 1[56]:[^:]_:Eastern Ewer:79FB:/ 209.4 "Slippery Soap" sync / 1[56]:[^:]_:Silkie:775E:/
> 
> So alternatively the previous trigger just needed ['Silkie', 'Eastern Ewer'].

I think this is a cleaner fix, I will implement this change.

File: ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts [hunk 2]
```diff
@@ -126,22 +126,31 @@ const triggerSet: TriggerSet<Data> = {
...
+      run: (data, matches) => data.suds = matches.effectId,
+    },
+    {
+      id: 'ASS Suds Lose',
+      // CE1 Bracing Suds (Wind / Donut)
+      // CE2 Chilling Suds (Ice / Cardinal)
+      // CE3 Fizzling Suds (Lightning / Intercardinal)
+      type: 'LosesEffect',
+      netRegex: { effectId: 'CE[1-3]', target: 'Silkie', capture: false },
+      run: (data, _matches) => delete data.suds,
```
Comment (Legends0) [inline ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts:144]: Can shorten this to:
`      run: (data) => delete data.suds,`

**Comment by wexxlee** [comment]
I have a user trigger that handles calling the puff that needs to be baited with the line (and which one is then safe to stand under) during slippery 1.  If I'm reading this file right, I think that's the first todo still remaining?  In any event, I'll wait until this PR is merged to avoid collisions before I clean it up and submit a new PR (probably not until Saturday).

File: ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts (hunk 2 above)
Comment (quisquous) [reply ui/raidboss/data/06-ew/dungeon/another_sildihn_subterrane.ts:144]: Here's a suggestion for easier committing

```suggestion
      run: (data) => delete data.suds,
```

**Comment by Legends0** [comment]
> I have a user trigger that handles calling the puff that needs to be baited with the line (and which one is then safe to stand under) during slippery 1. If I'm reading this file right, I think that's the first todo still remaining? In any event, I'll wait until this PR is merged to avoid collisions before I clean it up and submit a new PR (probably not until Saturday).

The todo was not updated, but this PR does add a call for the first puff. The other puffs it's going to just call avoid which actually I think can be wrong if it's a green puff after the rinse?
